cmake/modules/AddSwift.cmake

```diff
@@ -1531,6 +1531,11 @@ function(add_swift_library name)
             if("${sdk}" STREQUAL "ANDROID")
               list(APPEND swiftlib_private_link_libraries_targets
                    "-latomic")
+            # the same issue on FreeBSD, missing symbols:
+            # __atomic_store, __atomic_compare_exchange, __atomic_load
+            elseif("${sdk}" STREQUAL "FREEBSD")
+              list(APPEND swiftlib_private_link_libraries_targets
+                   "${SWIFTLIB_DIR}/clang/lib/freebsd/libclang_rt.builtins-${arch}.a")
```

Comment:
This seems like a very specific path to hardcode, but I'll take your word for it that it's a reasonable. (Especially for a CMake section we'd like to clean up later.)

Reply:
This hard-coded path does not work on my build. The `.a` file has been moved to `build/Ninja-ReleaseAssert/llvm-freebsd-x86_64/lib/clang/5.0.0/lib/freebsd/libclang_rt.builtins-${arch}.a`.

Reply:
Ah, `${SWIFTLIB_DIR}/clang` should be a symlink to `lib/clang/5.0.0`, so if you're seeing issues here, something else has gone wrong.

Reply:
Note that if you haven't built in a long time, the Clang version of the symlink may not match the actual Clang version.

Reply:
It's possible you just weren't using anything (from C) that needed atomics.